### PR TITLE
record: generalize export button

### PR DESCRIPTION
**Pull request by lauren-d**
* Adds export button component.

Co-Authored-by: Lauren-D <laurent.dubois@itld-solutions.be>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/export-button/export-button.component.html [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<ng-container *ngIf="exportOptions && exportOptions.length > 0">
+  <!-- check single export format -->
+  <ng-container *ngIf="exportOptions.length == 1; else  multipleFormats">
+    <!-- check if we can export list -->
+    <ng-container *ngIf="!exportOptions[0]?.disabled; else cannotExport">
+      <a id="search-export-button"
+         role="button" class="ml-2" [class]="cssClass"
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/export-button/export-button.component.html:24]: for me `ml-2` and all "display" classes should not be part of the core component.
It should be pass either using "cssClass", either on class of the component selection either using CSS (button + button selector)

File: projects/rero/ng-core/src/lib/record/export-button/export-button.component.html [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+         [href]="exportOptions[0].url">
+        <i class="fa fa-download mr-0 mr-sm-1"></i>
+        <span class="d-none d-sm-inline">{{ 'Export as' | translate }} {{ exportOptions[0].label }}</span>
+      </a>
+    </ng-container>
+    <ng-template #cannotExport>
+      <button id="search-export-button"
+              role="button"
+              class="ml-2 disabled"
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/export-button/export-button.component.html:33]: same

File: projects/rero/ng-core/src/lib/record/export-button/export-button.component.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+@Component({
+  selector: 'ng-core-export-button',
+  templateUrl: './export-button.component.html'
+})
+export class ExportButtonComponent {
+
+  /** Export formats configuration */
+  @Input()
+  exportOptions: {
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/export-button/export-button.component.ts:29]: single line ? ==> `@Input() exportOptions: { ... }`
This notation is used in all UI project and in many ng-core component (or it should be !)

**Review by zannkukai — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/export-button/export-button.component.html (hunk 1 above)
Comment (lauren-d) [reply projects/rero/ng-core/src/lib/record/export-button/export-button.component.html]: ok

File: projects/rero/ng-core/src/lib/record/export-button/export-button.component.ts (hunk 3 above)
Comment (lauren-d) [reply projects/rero/ng-core/src/lib/record/export-button/export-button.component.ts:29]: ok

File: projects/rero/ng-core/src/lib/record/export-button/export-button.component.html (hunk 2 above)
Comment (lauren-d) [reply projects/rero/ng-core/src/lib/record/export-button/export-button.component.html]: ok

**Review by jma — APPROVED** [review]

**Pull request merged**